apm-sniffer/optional-reporter-plugins/kafka-reporter-plugin/pom.xml

```diff
@@ -28,12 +28,21 @@
     <artifactId>kafka-reporter-plugin</artifactId>
     <packaging>jar</packaging>
 
+    <properties>
+        <shade.com.google.source>com.google</shade.com.google.source>
+    </properties>
+
     <dependencies>
         <dependency>
             <groupId>org.apache.kafka</groupId>
             <artifactId>kafka-clients</artifactId>
             <version>${kafka-clients.version}</version>
         </dependency>
+        <dependency>
```

Comment:
I don't think we need to explicitly add this dependency since we've already include agent-core

Reply:
This is old code. Similar to Gson, the kafka-reporter-plugin does not use any dependencies from apm-agent-core.

Reply:
> This is old code. Similar to Gson, the kafka-reporter-plugin does not use any dependencies from apm-agent-core.

IIRC, the shaded Gson library is already located in the agent-core jar. So just shading this library at compile-time is sufficient.

Reply:
> This is old code.

This is something you added, this is not old code.

Reply:
> > This is old code. Similar to Gson, the kafka-reporter-plugin does not use any dependencies from apm-agent-core.
> 
> IIRC, the shaded Gson library is already located in the agent-core jar. So just shading this library at compile-time is sufficient.
> 
> Just like what we've done for slf4j.

You are right, I will give it a try.